lib/Dialect/HW/HWAttributes.cpp

```diff
@@ -554,37 +554,67 @@ static Attribute simplifyShl(SmallVector<Attribute, 4> &operands) {
 
 static Attribute simplifyShrU(SmallVector<Attribute, 4> &operands) {
   assert(isHWIntegerType(operands[0].getType()));
-  // TODO: Implement support for identities like `x >> 0`.
+  // Implement support for identities like `x >> 0`.
+  if (auto rhs = operands[1].dyn_cast<IntegerAttr>()) {
+    if (rhs.getValue().isZero())
+      return operands[0];
+  }
```

Comment:
Super minor nit:  LLVM convention would be to leave off the outer brackets

Reply:
Thansk for reply.